torch/_dynamo/variables/lists.py

```diff
@@ -84,16 +84,29 @@ def call_method(
         if name == "__getitem__":
             assert not kwargs and len(args) == 1
             return self.getitem_const(args[0])
-        elif (
-            name == "__contains__"
-            and len(args) == 1
-            and args[0].is_python_constant()
-            and all(x.is_python_constant() for x in self.items)
-        ):
+        elif name == "__contains__":
+            assert len(args) == 1
             assert not kwargs
-            search = args[0].as_python_constant()
-            result = any(x.as_python_constant() == search for x in self.items)
-            return variables.ConstantVariable(result, **options)
+            search = args[0]
+            from .builtin import BuiltinVariable
+
+            check = [
+                BuiltinVariable(operator.eq).call_function(tx, [x, search], {})
+                for x in self.items
+            ]
+            if check_constant_args(check, {}):
+                result = any(c.as_python_constant() for c in check)
```

Comment:
This is o(n) (the initial list iter) + o(n) (check_constant_args) + o(up to n) (the any check or c in check).

Do you think we can get it to just o(up to n) by replacing it with a single iter over self.items? Up to you with what you think is best here.